### ISSUE TITLE
geomap geometry data fed from database not just geojson file

### ISSUE THREAD

**Issue by yosiasz** [grafana/grafana#82470]
**Why is this needed**:
1. Because currently you have to load geojson data from file on disk unlike any other plugin

2. The other issue is if the geojson file is big then it bogs down grafana and crashes.
Some can get up to 300mb ie -> 
https://geodata.bts.gov/datasets/7958468db586471d94f97e99b916175a_0/explore?location=34.666639%2C-108.315146%2C4.66

3. If someone wants to use a custom geojson file and use grafana cloud they might have to engage with grafana to copy geojson file(s) to the gazeteer folder for them?

**What would you like to be added**:
geomap geometry data fed from database not just geojson file

Ability to fetch geojson data from database. Orchestra Cities plugin (a fork of geomap) allows this

![image](https://github.com/grafana/grafana/assets/900951/2d1b4191-994c-4fa5-9f8a-38967b1c658c)

Data comes from sql server

![image](https://github.com/grafana/grafana/assets/900951/6d6d5ae8-07bf-4788-8c59-06efe620ed66)

**Who is this feature for?**

The world 🗺️ 🌍 

Thanks

**Comment by yosiasz** [comment]
the other issue is if the geojson file is big some can get up to 300mb ie
https://geodata.bts.gov/datasets/7958468db586471d94f97e99b916175a_0/explore?location=34.666639%2C-108.315146%2C4.66


then it bogs down grafana and makes it useless.

**Comment by yosiasz** [comment]
@drew08t here you go

**Comment by yosiasz** [comment]
@drew08t here it is my friend. appreciate it!

**Comment by yosiasz** [comment]
@Jayclifford345 here you go

**Comment by drew08t** [comment]
Hey @yosiasz, thanks for capturing this! I will discuss with the team to see when we can get this on the road map, although likely not until next quarter.

**Comment by kurt553** [comment]
This would be a great feature! This is a very common requirement of more complex and interesting map visualisations, and would be awesome to see what people would do with it. I'd also have a lot of immediate use for it

**Comment by yosiasz** [comment]
this is possible to do with orchestra cities plugin

**Comment by kurt553** [comment]
Thanks yeah. Even so, having it as inbuilt to Grafana removes the overhead of managing grafana cluster deployments/plugins and would allow for easy access to the functionality. It would be nice to have grafana team support this as a start toward having more complex and customisable map visualisation support for base grafana.

**Comment by yosiasz** [comment]
hello @drew08t any word on this feature request? i could ressurect my MR but tedious working in wsl on windows

**Comment by drew08t** [comment]
Hey @yosiasz, thanks for following up. Unfortunately, this is not on the roadmap for this quarter. I will bring it up for next quarter's planning.

**Comment by yosiasz** [comment]
@drew08t did this make it on the next roadmap? or maybe I can scavenge orchestra city and apply their changes and submit an MR.

**Comment by yosiasz** [comment]
will just use orchestra city for this.

**Comment by PeterCorless** [comment]
Would love to see if this would support Apache Pinot geospatial indexes and Grafana visualization. I believe there are IoT and observability use cases that could leverage geospatial data and visualization.

https://docs.pinot.apache.org/basics/indexing/geospatial-support
